### PR TITLE
feat: Support PodDisruptionBudgets

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/435

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: rust/operator/src/operations/pdb.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    #[snafu(display("Cannot apply PodDisruptionBudget [{name}]"))]
+    ApplyPdb {
+        source: stackable_operator::error::Error,
+        name: String,
+    },
+}
+
+pub async fn add_pdbs(
+    pdb: &PdbConfig,
+    zookeeper: &KafkaCluster,
```
Comment (sbernauer) [inline rust/operator/src/operations/pdb.rs:26]: marker to fix

**Comment by sbernauer** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/49/

**Comment by sbernauer** [comment]
Retry https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/50/

**Review by adwk67 — APPROVED** [review]
LGTM!

**Pull request merged**